tests/test_mig_shared_ssh.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+#
+# --- BEGIN_HEADER ---
+#
+# test_mig_shared_ssh - unit test of the corresponding mig shared module
+# Copyright (C) 2003-2024  The MiG Project by the Science HPC Center at UCPH
+#
+# This file is part of MiG.
+#
+# MiG is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# MiG is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests for the migrid module pointed to in the filename"""
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__))))
+
+from tests.support import TEST_OUTPUT_DIR, MigTestCase, FakeConfiguration, \
+    cleanpath, testmain
+from mig.shared.ssh import supported_pub_key_parsers, parse_pub_key, \
+    generate_ssh_rsa_key_pair
+
+
+class MigSharedSsh(MigTestCase):
+    """Wrap unit tests for the corresponding module"""
+
+    def test_ssh_key_generate_and_parse(self):
+        parsers = supported_pub_key_parsers()
+        # NOTE: should return a non-empty dict of algos and parsers
+        self.assertTrue(parsers)
+        self.assertTrue('ssh-rsa' in parsers)
+
+        # Generate common sized keys and parse the result
+        for keysize in (2048, 3072, 4096):
```

Comment:
If we are going to have test cases in a loop like this, it would be better to have something structured for this in the support library that can gather the individual successes/failures and then throw at the end of the test block informing the user of _all_ the failures. The thinking is that it would be rather frustrating to, say, fix the code for 3072 and then find the 4096 case was broken a different way.

Since looping over cases seems to be something that we want to reach for I'll think about what we can do to support this better.

Reply:
I totally agree - as we now also talked about in the meantime :-)